agent/task_context.go

```diff
@@ -519,3 +520,20 @@ func (tc *taskContext) getUserEndTaskResponse() *triggerEndTaskResp {
 
 	return tc.userEndTaskResp
 }
+
+func (tc *taskContext) deviceNames(ctx context.Context) error {
```

Comment:
nit: since this both gets the device names and populates the field, could we call this `getDeviceNames`/`populateDeviceNames` or something?

Reply:
done